### PR TITLE
runner: Add configurable way to block metadata instance server

### PR DESCRIPTION
**Pull request by doanac**
This allows us to block access the AWS/GCP instance metadata server running at http://169.254.169.254.

This service can leak information that CI jobs have no need to see.

### CODE REVIEW

**Comment by doanac** [comment]
ci-retest

**Comment by doanac** [comment]
ci-retest

File: jobserv_worker.py [hunk 1]
```diff
@@ -492,10 +492,28 @@ def _update_max_memory(rundef):
...
+            for line in f:
+                parts = line.split()
+                if parts[1].lower() == "fea9fea9":
+                    # 169.254.169.254 encoded
+                    flags = int(parts[3], base=8)
+                    if flags & 11:  # 11 - "unreachable route"
+                        log.warning("Route to cloud metadata service already blocked")
+                        return
+        log.warning("Blocking route to cloud metadata service")
+        subprocess.check_call(["ip", "route", "add", "unreachable", "169.254.169.254"])
```
Comment (vkhoroz) [inline jobserv_worker.py:509]: Don't we want to make the metadata service IP address configurable?

**Comment by doanac** [comment]
ci-retest

**Review by vkhoroz — COMMENTED** [review]
Are we blocking jobserv from accessing metadata server?
Or, are we blocking metadata service from accessing jobserv?

...I suspect it is the second use case, just confirming.

Also, can you describe how metadata service is accessing jobserv and what can be leaked?

File: jobserv_worker.py (hunk 1 above)
Comment (doanac) [reply jobserv_worker.py:509]: no. its a well-known value like 127.0.0.1 for cloud provider

Comment (doanac) [reply jobserv_worker.py:509]: it blocks a user from running something like `curl http://169.254.169.254/latest/meta-data/` in their CI job

Comment (doanac) [reply jobserv_worker.py:509]: i also just saw i need to block ipv6 `[fd00:ec2::254]`

Comment (doanac) [reply jobserv_worker.py]: accidentally pushed this to main. I did some cleanups. Let me know if you object and I'll fix.

Comment (vkhoroz) [reply jobserv_worker.py]: Whatever warks for you is fine.

**Pull request closed without merging**